### DISCUSSION TITLE
Nested context assign

### DISCUSSION THREAD

**Discussion by Ghirigoro** [statelyai/xstate#3724] [Q&A] [answered]
Currently it's not possible to assign to nested context like this:
```ts
assign({a:{b:1}});
```
I know that you can work around this with assignment functions, but I was curious why this limitation? Is it a technical issue (e.g. too expensive, difficult to implement, etc.), or just something that doesn't seem worth implementing - the juice is not worth the squeeze if you have existing workarounds?

**Comment by Andarist** [comment, marked as answer]
It's not what you always want. A lot of the time it's useful to wipe out the stale data and assign a completely new object that might have some optional properties. Merging deeply with the new data would effectively make that very quirky as you'd have to do stuff like `{ id: '123', name: 'Andarist', email: undefined }` - even if your new object looks like this: `{ id: '123', name: 'Andarist' }`. 

The current API is more flexible as it gives you the choice to merge things deeply or to replace them. To merge things deeply you'd have to do this though:
```ts
assign({
  a: ctx => ({ ...ctx.a, b: 1 })
})
```

**Reply by Ghirigoro** [reply]
Thanks. Appreciate the explanation.
